balancer/controllers/chatgpt.py

```diff
@@ -0,0 +1,80 @@
+from django.http import JsonResponse
+from bs4 import BeautifulSoup
+from nltk.stem import PorterStemmer
+import requests
+import openai
+import tiktoken
+import os
+import json
+
+# remove before production
+from django.views.decorators.csrf import csrf_exempt
+
+# summarize web content extracted from url
+
+
+@csrf_exempt
+def extract_webpage(request):
+    openai.api_key = os.getenv('OPENAI_API_KEY')
```

Comment:
Can you open an issue to add this key to setup? Given this is an open source project, I'll likely add a requirement to make that people provide their own key as an env variable before launching the service